### PR TITLE
Add support for filter alerts

### PR DESCRIPTION
**Pull request by jswoods**

### CODE REVIEW

File: examples/humiofilteralert.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  actions:
+    - example-email-action
+---
+apiVersion: core.humio.com/v1alpha1
+kind: HumioAction
+metadata:
+  name: example-email-action-external
+spec:
+  externalClusterName: example-humioexternalcluster
+name: example-filter-alert
```
Comment (SaaldjorMike) [inline examples/humiofilteralert.yaml:21]:
```suggestion
  name: example-filter-alert
```

File: examples/humiofilteralert.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name: example-filter-alert
+  viewName: humio
+  queryString: "#repo = humio | error = true"
+  enabled: true
+  description: Error counts
+  actions:
+    - example-email-action
+---
+apiVersion: core.humio.com/v1alpha1
+kind: HumioAction
```
Comment (SaaldjorMike) [inline examples/humiofilteralert.yaml:16]:
```suggestion
kind: HumioFilterAlert
```

Maybe we could also get https://github.com/humio/humio-operator/blob/master/examples/humioalert.yaml#L21 updated to say `kind: HumioAlert`? I assume that's the file that was the base of this new file, but that base file doesn't seem right.

File: controllers/suite/resources/humioresources_controller_test.go [hunk 3]
```diff
@@ -2237,6 +2237,180 @@ var _ = Describe("Humio Resources Controllers", func() {
...
+
+			toCreateDependentAction := &humiov1alpha1.HumioAction{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      actionKey.Name,
+					Namespace: actionKey.Namespace,
+				},
+				Spec: dependentEmailActionSpec,
+			}
+
+			suite.UsingClusterBy(clusterKey.Name, "HumioAlert: Creating the action required by the filter alert successfully")
```
Comment (SaaldjorMike) [inline controllers/suite/resources/humioresources_controller_test.go:2267]:
```suggestion
			suite.UsingClusterBy(clusterKey.Name, "HumioFilterAlert: Creating the action required by the filter alert successfully")
```

File: controllers/suite/resources/humioresources_controller_test.go [hunk 4]
```diff
@@ -2237,6 +2237,180 @@ var _ = Describe("Humio Resources Controllers", func() {
...
+
+			toCreateFilterAlert := &humiov1alpha1.HumioFilterAlert{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      key.Name,
+					Namespace: key.Namespace,
+				},
+				Spec: filterAlertSpec,
+			}
+
+			suite.UsingClusterBy(clusterKey.Name, "HumioAlert: Creating the filter alert successfully")
```
Comment (SaaldjorMike) [inline controllers/suite/resources/humioresources_controller_test.go:2300]:
```suggestion
			suite.UsingClusterBy(clusterKey.Name, "HumioFilterAlert: Creating the filter alert successfully")
```

**Review by SaaldjorMike — APPROVED** [review]

File: examples/humiofilteralert.yaml (hunk 2 above)
Comment (jswoods) [reply examples/humiofilteralert.yaml:16]: Yes, will fix.

**Pull request merged**